crates/ruff/src/commands/format.rs

```diff
@@ -213,7 +213,11 @@ pub(crate) fn format(
     match mode {
         FormatMode::Write => {
             if errors.is_empty() {
-                Ok(ExitStatus::Success)
+                if cli.exit_non_zero_on_format && results.any_formatted() {
+                    Ok(ExitStatus::Failure)
+                } else {
+                    Ok(ExitStatus::Success)
+                }
```

Comment:
The `check` command has a `--exit-zero` option that applies to all commands. I'd prefer this over having a `format` specific exit flag but I haven't verified if that satisfies all needs in the original issue

https://github.com/astral-sh/ruff/blob/75015b0ed91b728f22897c3bf12af5e6451e163a/crates/ruff/src/lib.rs#L448-L478